src/container/lookup.ts

```diff
@@ -1,7 +1,7 @@
 import * as ERROR_MSGS from "../constants/error_msgs";
 import { interfaces } from "../interfaces/interfaces";
 
-class Lookup<T extends interfaces.Clonable<T>> implements interfaces.Lookup<T> {
+class Lookup<T extends interfaces.Clonable<any> | any> implements interfaces.Lookup<T> {
```

Comment:
Maybe we can just remove the constraint:

```suggestion
class Lookup<T> implements interfaces.Lookup<T> {
```

Reply:
agree